system/security/shared/seclib.hpp

```diff
@@ -205,62 +228,52 @@ enum authStatus : int
+static const SecFeatureBit SUF_GetLastName                 = 0x0000000000000040;
+static const SecFeatureBit SUF_SetLastName                 = 0x0000000000000080;
+static const SecFeatureBit SUF_GetEmployeeID                 = 0x0000000000000100;
+static const SecFeatureBit SUF_SetEmployeeID                 = 0x0000000000000200;
+static const SecFeatureBit SUF_GetEmployeeNumber             = 0x0000000000000400;
+static const SecFeatureBit SUF_SetEmployeeNumber             = 0x0000000000000800;
+static const SecFeatureBit SUF_GetDistinguishedName        = 0x0000000000001000;
+static const SecFeatureBit SUF_SetDistinguishedName        = 0x0000000000002000;
+static const SecFeatureBit SUF_GetRealm                    = 0x0000000000004000;
+static const SecFeatureBit SUF_SetRealm                    = 0x0000000000008000;
+static const SecFeatureBit SUF_GetFqdn                       = 0x0000000000010000;
+static const SecFeatureBit SUF_SetFqdn                       = 0x0000000000020000;
+static const SecFeatureBit SUF_GetPeer                       = 0x0000000000040000;
+static const SecFeatureBit SUF_SetPeer                       = 0x0000000000080000;
+static const SecFeatureBit SUF_GetStatus                   = 0x0000000000100000;
+static const SecFeatureBit SUF_SetStatus                   = 0x0000000000200000;
+static const SecFeatureBit SUF_GetAuthenticateStatus       = 0x0000000000400000;
+static const SecFeatureBit SUF_SetAuthenticateStatus       = 0x0000000000800000;
+static const SecFeatureBit SUF_Credentials                   = 0x0000000001000000;
+static const SecFeatureBit SUF_GetUserID                     = 0x0000000002000000;
+static const SecFeatureBit SUF_CopyTo                        = 0x0000000004000000;
+static const SecFeatureBit SUF_GetPasswordExpiration         = 0x0000000008000000;
+static const SecFeatureBit SUF_SetPasswordExpiration       = 0x0000000010000000;
+static const SecFeatureBit SUF_GetPasswordDaysRemaining    = 0x0000000020000000;
+static const SecFeatureBit SUF_GetProperty                 = 0x0000000040000000;
+static const SecFeatureBit SUF_SetProperty                 = 0x0000000080000000;
+static const SecFeatureBit SUF_GetPropertyInt                = 0x0000000100000000;
+static const SecFeatureBit SUF_SetPropertyInt                = 0x0000000200000000;
+static const SecFeatureBit SUF_GetPropertyIterator           = 0x0000000400000000;
+static const SecFeatureBit SUF_Clone                         = 0x0000000800000000;
+static const SecFeatureBit SUF_GetDataElement              = 0x0000001000000000;
+static const SecFeatureBit SUF_GetDataElements             = 0x0000002000000000;
+static const SecFeatureSet SUF_ALL_FEATURES                = 0x0000003FFFFFFFFF; // update to include all added feature bits
 
 class CDateTime;
 interface IPropertyIterator;
 interface IPropertyTreeIterator;
-interface ISecUser : extends IInterface
+interface ISecUser : extends IInterface, implements ISecFeatureChecks
```

Comment:
@timothyklemm Not a huge fan of multiple inheritance unless you really are implementing interfaces that are independent.  Should ISecFeatureChecks derive from IInterface and these others derive only from ISecFeatureChecks?

Reply:
My intent was to define the interface in a way that it could be mixed in anywhere, without worrying about whether IInterface was already inherited through a different path. I wanted to avoid needing to use virtual inheritance. I don't have a specific use case in mind where this is an issue; it was more of a preventative measure.

I wonder if, when used as "the" base for different objects, it should be renamed to something more abstract (along the lines of ISecEntity). The current name is meant to convey the addition of one specific behavior, instead of being an umbrella of (potentially more) common behaviors.

Reply:
The general rule is that all interface should be directly or indirectly derived from IInterface.  If you want it to be "contained" then I would have a function return it rather than use multiple inheritance.
To keep things clean I would usually say an interface class can inherit one interface, and object class can inherit one object and multiple interfaces.
I agree that when thinking of this as the common base interface the name could be better.
Could be as simple as ISecInterface or ISecObject.  Depends a bit on what else we can envision ever going in there.